content-src/components/DiscoveryStreamComponents/DSImage/DSImage.jsx

```diff
@@ -0,0 +1,95 @@
+import {cache} from "./cache";
+import React from "react";
+import ReactDOM from "react-dom";
+
+export class DSImage extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isSeen: false,
+    };
+  }
+
+  onSeen(element) {
+    if (this.state && document.visibilityState === `visible` && element[0].isIntersecting) {
+      if (this.props.optimize) {
+        this.setState({
+          containerWidth: ReactDOM.findDOMNode(this).clientWidth,
+        });
+      }
+
+      this.setState({
+        isSeen: true,
+      });
+    }
+  }
+
+  reformatImageURL(url, width) {
+    const urlIsEncoded = url !== decodeURI(url);
+
+    // Encode the URL if it needs it
+    let constructedURL = urlIsEncoded ? url : encodeURIComponent(url);
+
+    // Change the image URL to request a size tailored for the parent container width
+    // Also: force JPEG, quality 60, no upscaling, no EXIF data
+    // Uses Thumbor: https://thumbor.readthedocs.io/en/latest/usage.html
+    constructedURL = `https://pocket-image-cache.com/${width}x0/filters:format(jpeg):quality(60):no_upscale():strip_exif()/${(constructedURL)}`;
+
+    // Use Mozilla CDN:
+    return `https://img-getpocket.cdn.mozilla.net/direct?url=${encodeURIComponent(constructedURL)}`;
+  }
+
+  componentDidMount() {
+    let options = {
+      root: document.querySelector(`document`),
+      threshold: 1,
```

Comment:
I think you want this to be 0.

what this does is saying how much of the element we're observing needs to be displayed before it is considered displayed. A 1 in this value is 100%, so if I scroll down and 1% of an image is still not fully scrolled, I see mostly a blank image, at 0, it waits until even a tiny bit of the image is in view before it's displayed.

Reply:
I dropped to 0.5, but we can adjust if that still feels like too much.

Reply:
I was initially thinking we do this asap. With 0.5 we just need to be aware that if someone scrolls down and sees just under half an image, it appears blank still. Might be more noticeable in cases where the screen height is low enough and the resolution is low enough that the page loads with < than 50% of an image visible.

What thoughts do you have on the pros and cons between 0.5 and 0?

Reply:
I just tried 0 and yeah, it's fine so I'll just change it! :D